2024-07-11 12:35:13.071650: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:35:13.071844: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:35:21,616 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:35:21,716 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:35:21,716 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:35:21,716 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:35:23,001 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:35:23,001 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,018 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.44}
INFO flwr 2024-07-11 12:35:23,018 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:35:23,255 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:35:23,277 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:35:23,278 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,286 | server.py:131 | fit progress: (1, 0.4089082453175707, {'accuracy': 0.7966666666666666}, 0.26872069999999937)
DEBUG flwr 2024-07-11 12:35:23,286 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:35:23,311 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:35:23,311 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:35:23,312 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:35:23,330 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,342 | server.py:131 | fit progress: (2, 0.41422950445984325, {'accuracy': 0.7933333333333333}, 0.32442579999999843)
DEBUG flwr 2024-07-11 12:35:23,342 | server.py:178 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:35:23,359 | server.py:192 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2024-07-11 12:35:23,359 | server.py:227 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:35:23,388 | server.py:241 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,399 | server.py:131 | fit progress: (3, 0.41717639328774647, {'accuracy': 0.79}, 0.3815528999999991)
DEBUG flwr 2024-07-11 12:35:23,399 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:35:23,419 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:35:23,419 | server.py:227 | fit_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,448 | server.py:241 | fit_round 4 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,461 | server.py:131 | fit progress: (4, 0.4205621466634243, {'accuracy': 0.7833333333333333}, 0.4430481999999998)
DEBUG flwr 2024-07-11 12:35:23,461 | server.py:178 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,484 | server.py:192 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:35:23,484 | server.py:227 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,513 | server.py:241 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,525 | server.py:131 | fit progress: (5, 0.42056642833356744, {'accuracy': 0.7833333333333333}, 0.5083021999999993)
DEBUG flwr 2024-07-11 12:35:23,526 | server.py:178 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,548 | server.py:192 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:35:23,548 | server.py:227 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,579 | server.py:241 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,591 | server.py:131 | fit progress: (6, 0.4205671168162418, {'accuracy': 0.7833333333333333}, 0.5734278999999987)
DEBUG flwr 2024-07-11 12:35:23,591 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,616 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:35:23,617 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,645 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,657 | server.py:131 | fit progress: (7, 0.42056719086728955, {'accuracy': 0.7833333333333333}, 0.6398219999999988)
DEBUG flwr 2024-07-11 12:35:23,658 | server.py:178 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,683 | server.py:192 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:35:23,684 | server.py:227 | fit_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,718 | server.py:241 | fit_round 8 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,730 | server.py:131 | fit progress: (8, 0.42056719884239757, {'accuracy': 0.7833333333333333}, 0.7117977999999994)
DEBUG flwr 2024-07-11 12:35:23,730 | server.py:178 | evaluate_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:35:23,758 | server.py:192 | evaluate_round 8 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:35:23,759 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:35:23,797 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,811 | server.py:131 | fit progress: (9, 0.415415567476925, {'accuracy': 0.79}, 0.7940014999999985)
DEBUG flwr 2024-07-11 12:35:23,812 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:35:23,840 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:35:23,841 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:35:23,882 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:35:23,895 | server.py:131 | fit progress: (10, 0.41544167847367186, {'accuracy': 0.79}, 0.8774163000000001)
DEBUG flwr 2024-07-11 12:35:23,895 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:35:23,936 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:35:23,936 | server.py:153 | FL finished in 0.9184181999999996
INFO flwr 2024-07-11 12:35:23,937 | app.py:225 | app_fit: losses_distributed [(1, 0.4070898238569498), (2, 0.40515660494565964), (3, 0.4106303205092748), (4, 0.3996544870592299), (5, 0.3996545019603911), (6, 0.39965451828071047), (7, 0.3996545140232359), (8, 0.3996545140232359), (9, 0.40328383701188225), (10, 0.4032840664897646)]
INFO flwr 2024-07-11 12:35:23,937 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:35:23,937 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:35:23,937 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.4089082453175707), (2, 0.41422950445984325), (3, 0.41717639328774647), (4, 0.4205621466634243), (5, 0.42056642833356744), (6, 0.4205671168162418), (7, 0.42056719086728955), (8, 0.42056719884239757), (9, 0.415415567476925), (10, 0.41544167847367186)]
INFO flwr 2024-07-11 12:35:23,937 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.44), (1, 0.7966666666666666), (2, 0.7933333333333333), (3, 0.79), (4, 0.7833333333333333), (5, 0.7833333333333333), (6, 0.7833333333333333), (7, 0.7833333333333333), (8, 0.7833333333333333), (9, 0.79), (10, 0.79)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.10037705,  1.11260027,  0.7423315 ,  0.25038145,  0.0997521 ,
         0.00960132, -0.03545998,  0.02165349]]), array([0.34884682])]
[array([[-2.10037705,  1.11260027,  0.7423315 ,  0.25038145,  0.0997521 ,
         0.00960132, -0.03545998,  0.02165349]]), array([0.34884682])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96813353,  1.11957142,  0.81727302,  0.30065556,  0.13804123,
         0.02660163,  0.0413835 ,  0.04539429]]), array([-0.06060537])]
[array([[-1.96813353,  1.11957142,  0.81727302,  0.30065556,  0.13804123,
         0.02660163,  0.0413835 ,  0.04539429]]), array([-0.06060537])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94903079,  1.02196476,  0.78246699,  0.35143456,  0.14407473,
         0.06928648,  0.04210744,  0.07052195]]), array([-0.06598279])]
[array([[-1.94903079,  1.02196476,  0.78246699,  0.35143456,  0.14407473,
         0.06928648,  0.04210744,  0.07052195]]), array([-0.06598279])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03410752,  1.00812964,  0.78734946,  0.40415529,  0.16590658,
         0.1152562 ,  0.04003486,  0.06252052]]), array([-0.06741174])]
[array([[-2.03410752,  1.00812964,  0.78734946,  0.40415529,  0.16590658,
         0.1152562 ,  0.04003486,  0.06252052]]), array([-0.06741174])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03412983,  1.00813299,  0.78742114,  0.40423533,  0.1658827 ,
         0.11531501,  0.04006827,  0.06252002]]), array([-0.06771418])]
[array([[-2.03412983,  1.00813299,  0.78742114,  0.40423533,  0.1658827 ,
         0.11531501,  0.04006827,  0.06252002]]), array([-0.06771418])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03412632,  1.00813741,  0.78741985,  0.40424579,  0.16588587,
         0.11532762,  0.04007353,  0.06252289]]), array([-0.06776621])]
[array([[-2.03412632,  1.00813741,  0.78741985,  0.40424579,  0.16588587,
         0.11532762,  0.04007353,  0.06252289]]), array([-0.06776621])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03412606,  1.00813782,  0.78741957,  0.40424713,  0.16588608,
         0.11532899,  0.04007398,  0.06252308]]), array([-0.06777138])]
[array([[-2.03412606,  1.00813782,  0.78741957,  0.40424713,  0.16588608,
         0.11532899,  0.04007398,  0.06252308]]), array([-0.06777138])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03412604,  1.00813787,  0.78741954,  0.40424727,  0.16588611,
         0.11532913,  0.04007403,  0.06252311]]), array([-0.06777194])]
[array([[-2.03412604,  1.00813787,  0.78741954,  0.40424727,  0.16588611,
         0.11532913,  0.04007403,  0.06252311]]), array([-0.06777194])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.00738064,  0.98209596,  0.80373134,  0.37337295,  0.11798318,
         0.10974143, -0.03485007,  0.05931255]]), array([0.08400738])]
[array([[-2.00738064,  0.98209596,  0.80373134,  0.37337295,  0.11798318,
         0.10974143, -0.03485007,  0.05931255]]), array([0.08400738])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.00726552,  0.98207359,  0.80403718,  0.37366002,  0.11805081,
         0.11004964, -0.03463373,  0.05925138]]), array([0.08257216])]
[array([[-2.00726552,  0.98207359,  0.80403718,  0.37366002,  0.11805081,
         0.11004964, -0.03463373,  0.05925138]]), array([0.08257216])]
>>>>>>>>>>>>>run time:
0.5625s
